spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/render/TupleVisitor.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.relational.core.sql.render;
+
+import org.springframework.data.relational.core.sql.TupleExpression;
+import org.springframework.data.relational.core.sql.Visitable;
+
+/**
+ * Visitor for rendering tuple expressions.
+ *
+ * @author Jens Schauder
+ * @since 3.5
+ */
+class TupleVisitor extends TypedSingleConditionRenderSupport<TupleExpression> implements PartRenderer {
```

Comment:
Everytime I'm looking into visitors, I have to re-learn what I wrote once 🙈 😆

Reply:
Same. But if I know something that works as well and is easier to understand my brain hasn't told me about it yet.